IndicoV2.Abstractions/Submissions/Models/Submission.cs

```diff
@@ -5,27 +5,70 @@
 
 namespace IndicoV2.Submissions.Models
 {
-    public class SubmissionSs : ISubmission
+    public class Submission : ISubmission
     {
-        private readonly IListSubmissions_Submissions_Submissions _ssSubmission;
-        public SubmissionSs(IListSubmissions_Submissions_Submissions submission) => _ssSubmission = submission;
-        public int Id => _ssSubmission.Id ?? 0;
+         /// <summary>
+        /// Submission id.
+        /// </summary>
+        public int Id { get; set; }
+
+        /// <summary>
+        /// Submission status. See <c><see cref="SubmissionStatus"/></c>.
+        /// </summary>
+        public SubmissionStatus Status { get; set; }
 
-        public SubmissionStatus Status => ConvertFromSs();
+        /// <summary>
+        /// Dataset id.
+        /// </summary>
+        public int DatasetId { get; set; }
 
-        public int DatasetId => _ssSubmission.DatasetId ?? 0;
+        /// <summary>
+        /// Workflow id.
+        /// </summary>
+        public int WorkflowId { get; set; }
 
-        public int WorkflowId => _ssSubmission.WorkflowId ?? 0;
+        /// <summary>
+        /// Submission input file.
+        /// </summary>
+        public string InputFile { get; set; }
 
-        public string InputFile => _ssSubmission.InputFile;
+        /// <summary>
+        /// Submission name of input file.
+        /// </summary>
+        public string InputFilename { get; set; }
 
-        public string InputFilename => _ssSubmission.InputFilename;
+        /// <summary>
+        /// Submission result file.
+        /// </summary>
+        public string ResultFile { get; set; }
 
-        public string ResultFile => _ssSubmission.ResultFile;
+        /// <summary>
+        /// Is submission retrieved.
+        /// </summary>
+        public bool Retrieved { get; set; }
 
-        public bool Retrieved => _ssSubmission.Retrieved ?? throw new ArgumentException("Invalid value for retrieved received from call");
+        /// <summary>
+        /// Submission errors.
+        /// </summary>
+        public string Errors { get; set; }
+    }
 
-        public string Errors => _ssSubmission.Errors;
+    public class SubmissionSs : Submission
```

Comment:
I wonder if this should be Internal, since it's something consumers shouldn't need to use/worry about

Reply:
Making it internal is causing issues where it's used since we would have to import the classes into `IndicoV2/Submissions` folder and `IndicoV2.Submissions` namespace. I think this could be something we address later? (how to organize our models and where to put them in the directory)

Reply:
Yeah for sure. Thanks for trying it! :)